soda_machine.rb

```diff
@@ -1,18 +1,39 @@
+require_relative 'soda'
+
 class SodaMachine
   attr_reader :sodas, :cash
+  attr_accessor :current_inventory_count
+  
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
-  end
-
-  def current_inventory_count
+    @current_inventory_count = @sodas.length
```

Comment:
What if you sell a soda then ask for inventory count again? Since you're caching this value, and don't update it when you delete, it'll be out of date.